File: src/main/java/app/application/listener/YoutubePlaylistDownloadListener.java
package app.application.listener;

import app.application.utils.DialogManager;
import app.application.utils.GlobalObjectHandler;
import javafx.application.Platform;

import java.io.File;

public class YoutubePlaylistDownloadListener extends YoutubeDownloadListener{

    private final int videoAmount;

    private final GlobalObjectHandler globalObjectHandler;

    private int actualVideo = 0;

    public YoutubePlaylistDownloadListener(DialogManager dialogManager, int videoAmount, GlobalObjectHandler globalObjectHandler) {
        super(dialogManager);
        this.videoAmount = videoAmount;
        this.globalObjectHandler = globalObjectHandler;
    }

    @Override
    public void onDownloading(int progress) {
        // Wird nicht verwendet
    }

    @Override
    public void onFinished(File data) {
        actualVideo ++;
        if(actualVideo == videoAmount){
            globalObjectHandler.getHostServices().showDocument(data.getParent());
            Platform.runLater(() -> dialogManager.openInformationDialog("Download fertig.", ""));
        }
    }

    @Override
    public boolean isDownloadFinished() {
        return false;
    }
}

File: src/main/kotlin/app/application/listener/YoutubePlaylistDownloadListener.kt
package app.application.listener

import app.application.spring.service.DialogManager
import app.application.spring.service.GlobalObjectHandler
import javafx.application.Platform
import java.io.File

class YoutubePlaylistDownloadListener(dialogManager: DialogManager, private val videoAmount: Int, private val globalObjectHandler: GlobalObjectHandler) : YoutubeDownloadListener(dialogManager) {
    private var actualVideo = 0
    override fun onDownloading(progress: Int) {
        // Not used
    }

    override fun onFinished(data: File) {
        actualVideo++
        if (actualVideo == videoAmount) {
            globalObjectHandler.hostServices?.showDocument(data.parent)
            Platform.runLater { dialogManager.openInformationDialog("Download finished.", "") }
        }
    }

    override val isDownloadFinished: Boolean
        get() = false

}